include/ruby/internal/intern/error.h

```diff
@@ -190,7 +190,6 @@ RBIMPL_ATTR_NONNULL(())
  */
 void rb_error_frozen(const char *what);
 
-RBIMPL_ATTR_NORETURN()
```

Comment:
(Picking on this line because it was what made me start investigating)

The only other caller of this is in `vm_setinstancevariable`, which does this:

```
    if (RB_SPECIAL_CONST_P(obj)) {
        rb_error_frozen_object(obj);
        return;
    }
```

I _think_ we want to get rid of that `return` (it was previously unreachable before this PR). Otherwise, I think setting an instance variable on a chilled string will emit the warning but not actually set the ivar, if I'm reading this correctly.

(I guess there should be a test case for setting ivars on chilled strings - the yjit bootstraptest has one but it's not actually checking that the ivar stuck)

Reply:
> Otherwise, I think setting an instance variable on a chilled string will emit the warning but not actually set the ivar, if I'm reading this correctly.

`RB_SPECIAL_CONST_P` is all immediates (`nil/false/true/Integer/Float`), so chilled strings wouldn't enter this branch. But it's a good catch regardless.

Reply:
I added a test for that.